neqo-bin/src/client/http09.rs

```diff
@@ -86,10 +87,33 @@ impl super::Handler for Handler<'_> {
                     }
                 }
                 ConnectionEvent::StateChange(
-                    State::WaitInitial | State::Handshaking | State::Connected,
+                    State::WaitInitial | State::Handshaking | State::Connected | State::Confirmed,
                 ) => {
                     qdebug!("{event:?}");
                     self.download_urls(client);
+                    if event == ConnectionEvent::StateChange(State::Confirmed) {
+                        if let Some((local_port, migration_addr)) = &self.migration {
+                            let mut local_addr = unspecified_addr(migration_addr);
+                            local_addr.set_port(*local_port);
+                            qdebug!("Migrating path to {:?} -> {:?}", local_addr, migration_addr);
+                            client
+                                .migrate(
+                                    Some(local_addr),
+                                    Some(*migration_addr),
+                                    false,
+                                    Instant::now(),
+                                )
+                                .map(|()| {
+                                    qinfo!(
+                                        "Connection migrated to {:?} -> {:?}",
+                                        local_addr,
+                                        migration_addr
+                                    );
+                                    // Don't do another migration.
+                                    self.migration = None;
```

Comment:
So you are only doing this change if the migration works.  What errors are you trying to ignore?  Would this be easier if the migration attempt only occurred for `State::Confirmed`?

I was going to suggest that you do `self.migration.take()` above instead, before I realized that migration could be attempted multiple times.

Reply:
You shouldn't need to call download_urls() when you hit Confirmed.  You always hit Connected first.